src/Toolkit/A.php

```diff
@@ -351,6 +351,31 @@ public static function has(
 		return in_array($value, $array, $strict);
 	}
 
+	/**
+	 * Join array elements as a string,
+	 * also supporting nested arrays
+	 */
+	public static function implode(
+		array $array,
+		string $separator = ', '
+	): string {
+		$result = '';
+
+		foreach ($array as $value) {
+			if (empty($result) === false) {
+				$result .= $separator;
+			}
+
+			if (is_array($value) === true) {
+				$result .= '[' . static::implode($value, $separator) . ']';
```

Comment:
I'm not sure about the bracket syntax TBH. It shifts the method from a pure implode method somewhat to a serialization method. From the name I'd expect to get a linear list of values. That would be achieved by:

```suggestion
				$result .= static::implode($value, $separator);
```

Would that be a problem for the validation use case where you use the new implode method?

Reply:
It could be a bit confusing, but it's not a real-world problem anyway. So I'm fine either way.